tests/beaker_tests/cisco_portchannel_global/test_portchannel_global.rb

```diff
@@ -0,0 +1,302 @@
+# 1 - errors have occurred,
+# 2 - changes have occurred
+# 4 - failures have occurred and
+# 6 - changes and failures have occurred.
+#
+# NOTE: 0 is the default exit_code checked in Beaker::DSL::Helpers::on() method.
+#
+# The test cases use RegExp pattern matching on stdout or output IO
+# instance attributes to verify resource properties.
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# -----------------------------
+# Common settings and variables
+# -----------------------------
+testheader = 'Resource cisco_portchannel_global'
+
+# Define PUPPETMASTER_MANIFESTPATH.
+
+# The 'tests' hash is used to define all of the test data values and expected
+# results. It is also used to pass optional flags to the test methods when
+# necessary.
+
+# 'tests' hash
+# Top-level keys set by caller:
+# tests[:master] - the master object
+# tests[:agent] - the agent object
+# tests[:show_cmd] - the common show command to use for test_show_run
+#
+tests = {
+  master: master,
+  agent:  agent,
+}
+
+# tests[id] keys set by caller and used by test_harness_common:
+#
+# tests[id] keys set by caller:
+# tests[id][:desc] - a string to use with logs & debugs
+# tests[id][:manifest] - the complete manifest, as used by test_harness_common
+# tests[id][:resource] - a hash of expected states, used by test_resource
+# tests[id][:resource_cmd] - 'puppet resource' command to use with test_resource
+# tests[id][:show_pattern] - array of regexp patterns to use with test_show_cmd
+# tests[id][:ensure] - (Optional) set to :present or :absent before calling
+# tests[id][:code] - (Optional) override the default exit code in some tests.
+#
+# These keys are local use only and not used by test_harness_common:
+#
+# tests[id][:manifest_props] - This is essentially a master list of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the list
+# tests[id][:resource_props] - This is essentially a master hash of properties
+#   that permits re-use of the properties for both :present and :absent testing
+#   without destroying the hash
+# tests[id][:title_pattern] - (Optional) defines the manifest title.
+#   Can be used with :af for mixed title/af testing. If mixing, :af values will
+#   be merged with title values and override any duplicates. If omitted,
+#   :title_pattern will be set to 'id'.
+#
+
+tests['default_properties_asym'] = {
+  title_pattern:  'default',
+  manifest_props: "
+    asymmetric                   => 'default',
+    bundle_hash                  => 'default',
+    bundle_select                => 'default',
+    hash_distribution            => 'default',
+    load_defer                   => 'default',
+    rotate                       => 'default',
+  ",
+  code:           [0, 2],
+  resource_props: {
+    'asymmetric'        => 'false',
+    'bundle_hash'       => 'ip',
```

Comment:
To validate the 10 or so `bundle_hash` attributes you can modify the existing `tests` hash like this.

```
tests["default_properties_asym"][:manifest_props]["bundle_hash"] = "ip-l4port"
tests["default_properties_asym"][:resource_props]["bundle_hash"] = "ip-l4port"
```
So.. you could loop through all possible values, modify the tests hash for only the values you want to re-test.